### ISSUE TITLE
Error: Cannot delete policy root: context deadline exceeded

### ISSUE THREAD

**Issue by aanm** [cilium/cilium#570]
Some times policy operations result in a `context deadline exceeded`. Some of the `runtime-tests` are getting this but not always reproducible.

For now I'll add `journalctl -u cilium` to see what happen in cilium in case tests fail

**Comment by aanm** [comment]
This might be caused by the `bpf/init.sh` script. For some unknown reason the scripts hangs in `tc filter replace dev ...`. @borkmann Is currently looking at the issue.

**Comment by aanm** [comment]
The message: `Error: Could not reach service1-instance1/private on port 80` is also related with this issue, since the daemon hangs because of `tc filter replace dev ...`, the endpoint is never build, and the test fails.

We don't realized it because jenkins exits after the error but if we run the test inside a while loop:
`while true; do ./11-getting-started.sh ; done;` we hit the `context deadline exceeded afterwards`

```
+ echo '------ performing HTTP GET on service1-instance1/private from service2 ------'
------ performing HTTP GET on service1-instance1/private from service2 ------
+ docker run --rm -i --net cilium-net -l id.service2 cilium/demo-client curl -si http://service1-instance1/private
+ abort 'Error: Could not reach service1-instance1/private on port 80'
+ set +x
------------------------------------------------------------------------
                            Test Failed
Error: Could not reach service1-instance1/private on port 80

------------------------------------------------------------------------
#kernel
IpInReceives                    3159               0.0
IpInDelivers                    3159               0.0
IpOutRequests                   3161               0.0
TcpActiveOpens                  420                0.0
TcpPassiveOpens                 210                0.0
TcpInSegs                       3178               0.0
TcpOutSegs                      3192               0.0
Ip6InReceives                   21                 0.0
Ip6InDelivers                   21                 0.0
Ip6OutRequests                  36                 0.0
Ip6OutDiscards                  1                  0.0
Ip6OutMcastPkts                 11                 0.0
Ip6InOctets                     42941              0.0
Ip6OutOctets                    18137              0.0
Ip6OutMcastOctets               1025               0.0
Ip6InNoECTPkts                  45                 0.0
Icmp6InMsgs                     2                  0.0
Icmp6OutMsgs                    9                  0.0
Icmp6InNeighborSolicits         1                  0.0
Icmp6InNeighborAdvertisements   1                  0.0
Icmp6OutNeighborSolicits        2                  0.0
Icmp6OutNeighborAdvertisements  1                  0.0
Icmp6OutMLDv2Reports            6                  0.0
Icmp6InType135                  1                  0.0
Icmp6InType136                  1                  0.0
Icmp6OutType135                 2                  0.0
Icmp6OutType136                 1                  0.0
Icmp6OutType143                 6                  0.0
Udp6OutDatagrams                3                  0.0
TcpExtTW                        2986               0.0
TcpExtDelayedACKLost            1                  0.0
TcpExtTCPHPHits                 424                0.0
TcpExtTCPPureAcks               1262               0.0
TcpExtTCPHPAcks                 11                 0.0
TcpExtTCPDSACKOldSent           1                  0.0
TcpExtTCPRcvCoalesce            313                0.0
TcpExtTCPAutoCorking            2                  0.0
TcpExtTCPOrigDataSent           1283               0.0
TcpExtTCPKeepAlive              3                  0.0
IpExtInOctets                   379186             0.0
IpExtOutOctets                  333326             0.0
IpExtInNoECTPkts                3159               0.0
./helpers.bash: line 25: kill: (30591) - No such process
service1-instance1
cilium-net
./helpers.bash: line 25: kill: (30591) - No such process
------ checking cilium status ------
Please run the monitor with root privileges.
Allocated IPv4 addresses:
 10.15.28.238
Allocated IPv6 addresses:
KVStore:            Ok
ContainerRuntime:   Ok
Kubernetes:         Disabled
Cilium:             Ok
------ creating Docker network of type Cilium ------
b7245e9a1001107e57ea89e40d6e294d4c945c315a0fb132b0e69eb73994116f
------ starting example service with Docker ------
2d7e9f7994302477d83531d8119deb5c4321904ebac8732c4cb7f53640047d6d
docker: Error response from daemon: failed to create endpoint service1-instance1 on network cilium-net: NetworkDriver.CreateEndpoint: Error retrieving endpoint context deadline exceeded.
service1-instance1
cilium-net
./helpers.bash: line 25: kill: (31509) - No such process
```

**Comment by borkmann** [comment]
Seen `Could not reach service1-instance1/private on port 80` also without `context deadline exceeded`. Potentially related could be issue #633.

**Comment by aanm** [comment]
@borkmann It's actually this one https://github.com/cilium/cilium/issues/629

**Comment by borkmann** [comment]
Fixed with #685 and #629.
